pkg/volume/csi/fake/fake_client.go

```diff
@@ -172,14 +175,29 @@ func (f *NodeClient) NodePublishVolume(ctx context.Context, req *csipb.NodePubli
 		return nil, timeoutErr
 	}
 
-	f.nodePublishedVolumes[req.GetVolumeId()] = CSIVolume{
+	// "Creation of target_path is the responsibility of the SP."
+	// Our plugin depends on it.
+	if req.VolumeCapability.GetBlock() != nil {
+		if err := ioutil.WriteFile(req.TargetPath, []byte{}, 0644); err != nil {
+			return nil, fmt.Errorf("cannot create target path %s for block file: %s", req.TargetPath, err)
+		}
+	} else {
+		if err := os.MkdirAll(req.TargetPath, 0755); err != nil {
+			return nil, fmt.Errorf("cannot create target directory %s for mount: %s", req.TargetPath, err)
+		}
+	}
+
+	publishedVolume := CSIVolume{
 		VolumeHandle:    req.GetVolumeId(),
 		Path:            req.GetTargetPath(),
 		DeviceMountPath: req.GetStagingTargetPath(),
 		VolumeContext:   req.GetVolumeContext(),
 		FSType:          req.GetVolumeCapability().GetMount().GetFsType(),
```

Comment:
You conditionally added `MountFlags`. Should `FSType` also be made conditional? (not worried about the nil check, just the logic.)

Or is the difference between empty string and nil `[]string` significant in the way we do `CSIVolume` objects?

Reply:
Fixed (moved FsType under `if GetMount() != nil`)